contracts/ImplementationLogic.sol

```diff
@@ -36,9 +37,13 @@ contract ImplementationLogic is EternalStorage, Ownable(msg.sender) {
         bytes memory data,
         uint256 gasLimit
     ) internal returns (bool success) {
+        require(
```

Comment:
This should be checked after the assembly code for initializing the txGas variable.. as that value will be the one going in the call() method.

Reply:
There are two approaches to take. 1) before call() 2) after call.() which approach are we taking?

I did it according to 'before call()',either gasLimit or 'gas' is being passed to call() and we put a require statement before the call to check the gasLimit.

Reply:
txGas uses either 'gasLimit' where we have require statement or 'gas' which is already == gasleft(). no point checking that right? since call is using every gas available.